From: Pino Toscano <toscano.pino@tiscali.it>
Date: Mon, 27 Aug 2018 10:08:15 +0200
Subject: Enable building on kFreeBSD

Make use of the "freebsd" OS implementation, as the kernel is the same.

Also, provide build fixes for task-manager-freebsd.c:
- NZERO is not provided by kfreebsd-kernel-headers, as already provided by
  limits.h but only when _XOPEN_SOURCE is requested (and it is not in this
  case); make use of sysconf to query that value instead
Last-Update: 2018-08-27
Forwarded: no
---
 configure.ac               | 2 +-
 src/task-manager-freebsd.c | 5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 16fcb83..8822f8b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -96,7 +96,7 @@ if test x"$ac_skel" = x"yes"; then
 	ac_os_implementation="skel"
 else
 	case "$target_os" in
-		freebsd*)
+		freebsd*|kfreebsd*-gnu)
 		ac_os_implementation="freebsd"
 		AC_CHECK_LIB([kvm], [kvm_openfiles])
 		AC_CHECK_HEADERS([fcntl.h kvm.h paths.h pwd.h sys/param.h sys/proc.h \
diff --git a/src/task-manager-freebsd.c b/src/task-manager-freebsd.c
index 3a96814..adf85fc 100644
--- a/src/task-manager-freebsd.c
+++ b/src/task-manager-freebsd.c
@@ -135,6 +135,7 @@ get_task_details (struct kinfo_proc *kp, Task *task)
 	char buf[1024], *p;
 	size_t bufsz;
 	int i, oid[4];
+	long nzero = sysconf (_SC_NZERO);
 
 	memset(task, 0, sizeof(Task));
 	task->pid = kp->ki_pid;
@@ -205,9 +206,9 @@ get_task_details (struct kinfo_proc *kp, Task *task)
 	i++;
 	if (!(kp->ki_sflag & PS_INMEM))
 		task->state[i++] = 'W';
-	if (kp->ki_nice < NZERO)
+	if (kp->ki_nice < nzero)
 		task->state[i++] = '<';
-	else if (kp->ki_nice > NZERO)
+	else if (kp->ki_nice > nzero)
 		task->state[i++] = 'N';
 	if (kp->ki_flag & P_TRACED)
 		task->state[i++] = 'X';
